### ISSUE TITLE
SparsityPattern doesn't initialize properly

### ISSUE THREAD

**Issue by QiaoLei-88** [QiaoLei-88/NSolver#120]
commit a88286d49ea56365e42013553afde76245f16fcf  breaks regTest d2_slide_case001003.

Error message is:

```
------------------------------------------
Got error 2 in row 1640 of proc 1 when trying to add the columns:
1640 1641 1642 1643 1644 1645 1646 1647

Matrix row (nonlocal part) has the following indices:
2396 2397 2398 2399 136 137 138 139 140 141 142 143 1640 1641 1642 1643 180 181 182 183 188 189 190 191


--------------------------------------------------------
An error occurred in line <1730> of file </u/qiaolei/devel/deal.II/dealii/source/lac/trilinos_sparse_matrix.cc> in function
    void dealii::TrilinosWrappers::SparseMatrix::add(dealii::TrilinosWrappers::SparseMatrix::size_type, dealii::TrilinosWrappers::SparseMatrix::size_type, const size_type*, const double*, bool, bool)
The violated condition was:
    ierr <= 0
The name and call sequence of the exception was:
    ExcAccessToNonPresentElement(row, col_index_ptr[0])
Additional Information:
You tried to access element (1640/1640) of a sparse matrix, but it appears to not exist in the Trilinos sparsity pattern.

Stacktrace:
-----------
#0  /u/qiaolei/Library/deal.II/lib/libdeal_II.g.so.8.4.0-pre: dealii::TrilinosWrappers::SparseMatrix::add(unsigned int, unsigned int, unsigned int const*, double const*, bool, bool)
#1  /u/qiaolei/devel/NSolver/build/bin/NSolver: NSFEMSolver::NSolver<2>::assemble_face_term(unsigned int, dealii::FEFaceValuesBase<2, 2> const&, dealii::FEFaceValuesBase<2, 2> const&, std::vector<unsigned int, std::allocator<unsigned int> > const&, std::vector<unsigned int, std::allocator<unsigned int> > const&, bool, unsigned int, double)
#2  /u/qiaolei/devel/NSolver/build/bin/NSolver: NSFEMSolver::NSolver<2>::assemble_system()
#3  /u/qiaolei/devel/NSolver/build/bin/NSolver: NSFEMSolver::NSolver<2>::run()
#4  /u/qiaolei/devel/NSolver/build/bin/NSolver: main
--------------------------------------------------------
```

**Comment by QiaoLei-88** [comment]
The cell caused this trouble is not even on subdomain boundary,
and the sparsity pattern is definitely wrong.

![bugfaceflux](https://cloud.githubusercontent.com/assets/10951883/11050553/a180b6b6-870c-11e5-9039-53d16b777b0b.png)

## 

Update.
The picture above is not correct. 
The showed subdomain ID is of last time step before re-partitioning.

**Comment by QiaoLei-88** [comment]
Fixed by 1a9ae48ae3236a05f097bc9ebb3286131f3b72a2
